tt_metal/impl/program/program.hpp

```diff
@@ -144,6 +144,9 @@ class Program {
     bool is_finalized() const { return this->finalized_; }
     void finalize();
 
+    void capture_multi_device_dependencies() { capture_multi_device_dependencies_ = true; }
```

Comment:
Maybe `set_multi_device_dependencies` or `has_multi_device_dependencies`? I confused myself when reading `program.capture_multi_device_dependencies();` because we talk about capture trace.